### PR TITLE
a8n: periodically ensure creation of bitbucket server webhook

### PR DESCRIPTION
**Pull request by kzh**
This PR periodically ensures the creation of webhooks on bitbucket server instances for automation usage.

### CODE REVIEW

File: enterprise/internal/a8n/webhooks.go [hunk 1]
```diff
@@ -412,6 +413,48 @@ func (*GitHubWebhook) pullRequestReviewCommentEvent(e *gh.PullRequestReviewComme
...
+				Name:     "sourcegraph-a8n",
+				Scope:    "global",
+				Events:   []string{"pr"},
+				Endpoint: endpoint,
+				Secret:   con.Webhooks.Secret,
+			}
+
+			err = client.EnsureWebhook(context.Background(), wh)
+			if err != nil {
+				log15.Error("Ensuring BBS Webhook failed [HTTP Request]", err.Error())
```
Comment (tsenart) [inline enterprise/internal/a8n/webhooks.go]:
```suggestion
				log15.Error("Ensuring BBS Webhook exists failed [HTTP Request]", err.Error())
```

File: enterprise/internal/a8n/webhooks.go [hunk 2]
```diff
@@ -412,6 +413,48 @@ func (*GitHubWebhook) pullRequestReviewCommentEvent(e *gh.PullRequestReviewComme
...
+		for _, e := range es {
+			c, _ := e.Configuration()
+			con, ok := c.(*schema.BitbucketServerConnection)
+			if !ok || con.Webhooks == nil {
+				continue
+			}
+
+			client, err := bbs.NewClientWithConfig(con)
+			if err != nil {
+				log15.Error("Ensuring BBS Webhook failed [Creating Client]", err.Error())
```
Comment (tsenart) [inline enterprise/internal/a8n/webhooks.go]:
```suggestion
				log15.Error("Ensuring BBS Webhook exists failed [Creating Client]", err.Error())
```

File: enterprise/internal/a8n/webhooks.go [hunk 3]
```diff
@@ -412,6 +413,48 @@ func (*GitHubWebhook) pullRequestReviewCommentEvent(e *gh.PullRequestReviewComme
 	return &comment
 }
 
+// EnsureCreation ensures the creation of the BitbucketServer automation webhook.
+func (h *BitbucketServerWebhook) EnsureCreation() {
+	for {
+		args := repos.StoreListExternalServicesArgs{Kinds: []string{"BITBUCKETSERVER"}}
+		es, err := h.Repos.ListExternalServices(context.Background(), args)
+		if err != nil {
+			log15.Error("Ensuring BBS Webhook failed [Listing BBS extsvc]", err.Error())
```
Comment (tsenart) [inline enterprise/internal/a8n/webhooks.go]:
```suggestion
			log15.Error("Ensuring BBS Webhook exists failed [Listing BBS extsvc]", err.Error())
```

File: enterprise/internal/a8n/webhooks.go [hunk 4]
```diff
@@ -412,6 +413,48 @@ func (*GitHubWebhook) pullRequestReviewCommentEvent(e *gh.PullRequestReviewComme
...
+			continue
+		}
+
+		for _, e := range es {
+			c, _ := e.Configuration()
+			con, ok := c.(*schema.BitbucketServerConnection)
+			if !ok || con.Webhooks == nil {
```
[the remark this answers is not in the document]
Comment (kzh) [reply enterprise/internal/a8n/webhooks.go]: `con.Webhooks` is a struct. I will change `Webhooks` -> `Webhook`.

**Pull request merged**